### ISSUE TITLE
error in ch 8

### ISSUE THREAD

**Issue by vasishth** [bnicenboim/bayescogsci#43]
> Next, create the model matrix, $X$, and prepare the data as a list. As in section \@ref(sec-interstan), exclude the intercept from the matrix `X` using ` 0 +... `. This is because the Stan code that we are using already takes into account that the first column in the model matrix is going to be a vector of ones.

I think this is incorrect. In section  \@ref(sec-interstan), it is not the case that that the Stan code ensures that the first column in the model matrix (which is X) is going to be a vector of ones. Rather, what is happening is that the intercept is kept separate from the model matrix X, which only has the contrast coding for the slope (beta) parameters. 

Can you correct this @bnicenboim ?

Or am I mistaken about the way the Stan code implements this in section  \@ref(sec-interstan)?

**Comment by bnicenboim** [comment]
yes, it's true that the intercept is kept separate from the model matrix X. I meant that this is "as if the first column in the model matrix is a vector of ones."

Can I write the following? 

This is because in the Stan code that we are using, the intercept is kept separate from the model matrix X, which only has the contrast coding for the slope parameters. This is as if the first column in the model matrix is already a vector of ones.

**Comment by vasishth** [comment]
I would edit the last line as follows:

>  The approach we take here is equivalent to defining a model matrix whose first column is a vector of ones. What would then change in the model is that the beta vector (which in our implementation only contains the slope parameters) would contain both the intercept and slope parameters. 

I felt that the "as if" is a bit confusing. Ideally, I would have just shown a toy example illustrating two ways to write the same model:

$ y = \alpha + X \beta + \varepsilon$

where X is

```
(X <- model.matrix(~ 0 + c_set_size * c_trial, df_recall))
```

Or we could write the model as

$ y = X \beta + \varepsilon$

Now, X is

```
(X <- model.matrix(~ 1 + c_set_size * c_trial, df_recall))
```

But maybe it's too late to be adding text to the book now.  So probably a minimal edit like the one you suggest is better. When teaching from the book, I always spell this stuff out in the lecture.

**Comment by bnicenboim** [comment]
I actually find this last part a bit confusing: "What would then change in the model is that the beta vector (which in our implementation only contains the slope parameters) would contain both the intercept and slope parameters."

I would just leave it like this then:

This is because in the Stan code that we are using, the intercept is kept separate from the model matrix $X$, which only has the contrast coding for the slope parameters. The approach we take here is equivalent to defining a model matrix whose first column is a vector of ones. 

ok?

**Comment by vasishth** [comment]
fixed.
